engine/include/Engine/Core/Rendering/Window/WindowGLFW.hpp

```diff
@@ -5,63 +5,63 @@
  */
 #pragma once
 
-#include "Engine/Core/Rendering/Window/IWindow.hpp"
+#include <string> //std::string
 
 class GLFWwindow;
 
 namespace Engine::Core::Renderering
 {
-class WindowGLFW final : public IWindow
+
+struct WindowCreateArg
 {
-protected:
-    GLFWwindow* m_window = nullptr;
+    std::string name = "";
+    unsigned short height = 0;
+    unsigned short width = 0;
+};
 
+class Window
+{
 protected:
-    void initialize(const WindowCreateArg& arg) noexcept final;
-    void release() noexcept final;
+    GLFWwindow* m_window = nullptr;
 
 public:
-    inline WindowGLFW(const WindowCreateArg& arg) noexcept : IWindow()
-    {
-        initialize(arg);
-    }
 
-    inline ~WindowGLFW() noexcept
-    {
-        release();
-    }
+    Window(const WindowCreateArg& arg) noexcept;
+
+    ~Window() noexcept;
 
-    WindowGLFW() noexcept = delete;
+    Window() noexcept = delete;
 
-    WindowGLFW(const WindowGLFW& other) noexcept = delete;
+    Window(const Window& other) noexcept = delete;
 
-    WindowGLFW(WindowGLFW&& other) noexcept = delete;
+    Window(Window&& other) noexcept = delete;
 
-    WindowGLFW& operator=(const WindowGLFW& other) noexcept = delete;
+    Window& operator=(const Window& other) noexcept = delete;
 
-    WindowGLFW& operator=(WindowGLFW&& other) noexcept = delete;
+    Window& operator=(Window&& other) noexcept = delete;
 
     /**
      * @brief return GLFWwindow pointer with generic forme that allow user to ealsy use another lib
      * @return void*
      */
-    [[nodiscard]] inline const void* get() const noexcept final;
+    [[nodiscard]] inline const void* get() const noexcept;
 
-    [[nodiscard]] inline void* get() noexcept final;
+    [[nodiscard]] inline void* get() noexcept;
 
     /**
      * @brief Get the glfw current window
      * @return
      */
-    [[nodiscard]] constexpr inline const GLFWwindow* const getGLFWWindow() const noexcept;
+    [[nodiscard]] constexpr inline const GLFWwindow* getGLFWWindow() const noexcept;
```

Comment:
The purpose of the Window class is to encapsulate the glfw code, and to not use GLFWwindow raw.
Is it necessary to define a public getter ? 
In the case of glfw, the getter is probably necessary because of inputs etc.
However, could a protected or private getter be better ? (with friends)

Reply:
Its an extrem solution to allow user to use feature of GLFW wrapping of this feature is not implemented